src/coreclr/src/vm/peimage.h

```diff
@@ -245,14 +246,17 @@ class PEImage
     {
 
         LPCWSTR m_pPath;
+        BOOL m_bIsInBundle;
 
-        PEImageLocator(LPCWSTR pPath)
-            : m_pPath(pPath)
+        PEImageLocator(LPCWSTR pPath, BOOL bIsInBundle)
+            : m_pPath(pPath),
+              m_bIsInBundle(bIsInBundle)
         {
         }
 
-        PEImageLocator(PEImage * pImage)
-            : m_pPath(pImage->m_path.GetUnicode())
+        PEImageLocator(PEImage * pImage, BOOL bIsInBundle)
+            : m_pPath(pImage->m_path.GetUnicode()),
+              m_bIsInBundle(bIsInBundle)
```

Comment:
Can this just initialize `m_bIsInBundle` using `pImage` - `IsInBundle()`? Or is there a reason we'd need to be able to use a different value?

Reply:
If we use `pImage->IsInBundle()` here then we'd have the problem that this PR is trying to fix - since we'd be allowing all bundled files to be located.

Whenever calling `AssemblyNative::LoadFromPath`, we'll call `PEImage::OpenImage` with an invalid bundle file location, which will eventually cause a locator to be instantiated with `m_bIsInBundle` set to false, even though `pImage->IsInBundle` can be true.

Reply:
Wouldn't that go through the other constructor that initializes using a path instead of a `PEImage` instance?

My interpretation of this constructor is that it should create a locator for the `PEImage` instance that is being used to initialize it, which would now also include whether or not that instance is in the bundle.

Reply:
Ah, yeah, you're right. I've changed this, thanks.